homeassistant/components/minio/__init__.py

```diff
@@ -0,0 +1,280 @@
+"""Minio component."""
+
+import logging
+import os
+import threading
+from queue import Queue
+from typing import List
+
+import voluptuous as vol
+from homeassistant.const import EVENT_HOMEASSISTANT_START, \
+    EVENT_HOMEASSISTANT_STOP
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'minio'
+CONF_HOST = 'host'
+CONF_PORT = 'port'
+CONF_ACCESS_KEY = 'access_key'
+CONF_SECRET_KEY = 'secret_key'
+CONF_SECURE = 'secure'
+CONF_LISTEN = 'listen'
+CONF_LISTEN_BUCKET = 'bucket'
+CONF_LISTEN_PREFIX = 'prefix'
+CONF_LISTEN_SUFFIX = 'suffix'
+CONF_LISTEN_EVENTS = 'events'
+
+CONF_LISTEN_PREFIX_DEFAULT = ''
```

Comment:
We usually only prefix config keys with `CONF_`. Default config values we prefix with `DEFAULT_`.